lib/mutate/Store/index.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import StoreItemList from "./StoreItemList";
+import StoreContainer from "./Container";
+import StoreSearchBar from "./StoreSearchBar";
+import StoreContentContainer from "./StoreContentContainer";
+import search from "../npm-search";
+import debounce from "debounce";
+
+class Store extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchValue: "",
+      items: []
+    };
+
+    // Don't search more than once every 100 miliseconds
+    this.searchAndUpdate = debounce(this.searchAndUpdate, 100);
+
+    this.searchAndUpdate();
+  }
+
+  searchAndUpdate = query => {
+    search(query)
+      .then(res => res.data)
```

Comment:
just curious here: why are there two `then`s?

Reply:
One then filters out the `data` attribute from the query. It's the same thing as:

``` js
search(query)
  .then(({ data }) => { 
    this.setState({ items: data.objects });
  });
```

Though I tend to find this more readable.

If your asking why you can attach a `then` off a `then`, it's because the function `then` [returns a Promise.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then)